### PR TITLE
Added s3 logs for pv and pvc upload

### PR DESCRIPTION
**Pull request by jmishrarh**
Added log  before and after uploading to s3 store. These logs will help to debug  zero byte VRG download issue. 

Fixes [issue](https://github.com/RamenDR/ramen/issues/847)

### CODE REVIEW

**Review by rakeshgm — APPROVED** [review]

File: controllers/vrg_volrep.go [hunk 1]
```diff
@@ -633,6 +633,8 @@ func (v *VRGInstance) UploadPVandPVCtoS3Store(s3ProfileName string, pvc *corev1.
 func (v *VRGInstance) UploadPVAndPVCtoS3(s3ProfileName string, objectStore ObjectStorer,
 	pv *corev1.PersistentVolume, pvc *corev1.PersistentVolumeClaim,
 ) error {
+	v.log.Info("About to upload PV to s3 profile", "PV", pv.Name, "s3 profile", s3ProfileName)
```
Comment (nirs) [inline controllers/vrg_volrep.go:636]: Maybe "Uploading PV to s3 store", "pv", pv.Name, "profile", s3ProfileName ?

File: controllers/vrg_volrep.go [hunk 2]
```diff
@@ -647,16 +649,22 @@ func (v *VRGInstance) UploadPVAndPVCtoS3(s3ProfileName string, objectStore Objec
 		return err
 	}
 
+	v.log.Info("Uploaded PV to s3 profile", "PV", pv.Name, "s3 profile", s3ProfileName)
```
Comment (nirs) [inline controllers/vrg_volrep.go:652]: Logging before and after an operation is important when the operation takes log of time, but these logs are debug logs. In INFO level logs it is good enough to before the operation. If the operation fails we should see an error log.

File: controllers/vrg_volrep.go [hunk 3]
```diff
@@ -633,6 +633,8 @@ func (v *VRGInstance) UploadPVandPVCtoS3Store(s3ProfileName string, pvc *corev1.
 func (v *VRGInstance) UploadPVAndPVCtoS3(s3ProfileName string, objectStore ObjectStorer,
 	pv *corev1.PersistentVolume, pvc *corev1.PersistentVolumeClaim,
 ) error {
+	v.log.Info("About to upload PV to s3 profile", "PV", pv.Name, "s3 profile", s3ProfileName)
+
 	if err := UploadPV(objectStore, v.s3KeyPrefix(), pv.Name, *pv); err != nil {
```
Comment (nirs) [inline controllers/vrg_volrep.go:638]: We can log inside UploadPV - this way all callers get logging and we cannot forget to log.

File: controllers/vrg_volrep.go (hunk 2 above)
Comment (jmishrarh) [reply controllers/vrg_volrep.go:652]: These are INFO logs. Each block has error logging. 
2023-11-08T21:23:29.214+0530  INFO    controllers.VolumeReplicationGroup.vrginstance  About to upload PV to s3 profile        {"VolumeReplicationGroup": "envtest-ns-sdixm/vrg-sdixm", "rid": "d5b10778-cf83-48ca-897c-d810ed18bea0", "State": "primary", "PV": "pv0001", "s3 profile": "s3profile0"}
  2023-11-08T21:23:29.214+0530  INFO    controllers.VolumeReplicationGroup.vrginstance  Uploaded PV to s3 profile       {"VolumeReplicationGroup": "envtest-ns-sdixm/vrg-sdixm", "rid": "d5b10778-cf83-48ca-897c-d810ed18bea0", "State": "primary", "PV": "pv0001", "s3 profile": "s3profile0"}
  2023-11-08T21:23:29.214+0530  INFO    controllers.VolumeReplicationGroup.vrginstance  About to upload PVC to s3 profile       {"VolumeReplicationGroup": "envtest-ns-sdixm/vrg-sdixm", "rid": "d5b10778-cf83-48ca-897c-d810ed18bea0", "State": "primary", "PVC": "pvc-of-pv0001", "s3 profile": "s3profile0"}
  2023-11-08T21:23:29.214+0530  INFO    controllers.VolumeReplicationGroup.vrginstance  Uploaded PVC to s3 profile      {"VolumeReplicationGroup": "envtest-ns-sdixm/vrg-sdixm", "rid": "d5b10778-cf83-48ca-897c-d810ed18bea0", "State": "primary", "PVC": "pvc-of-pv0001", "s3 profile": "s3profile0"}
  2023-11-08T21:23:29.216+0530  INFO    pvcmap.VolumeReplicationGroup   Update event for PersistentVolumeClaim
  2023-11-08T21:23:29.216+0530  INFO    pvcmap.VolumeReplicationGroup   Reconciling due to VR Protection finalizer      {"pvc": "envtest-ns-sdixm/pvc-of-pv0001"}
  2023-11-08T21:23:29.216+0530  INFO    pvcmap.VolumeReplicationGroup   Found VolumeReplicationGroup with matching labels       {"pvc": "envtest-ns-sdixm/pvc-of-pv0001", "vrg": "vrg-sdixm", "labeled": ""}
  2023-11-08T21:23:29.216+0530  INFO    pvcmap.VolumeReplicationGroup   Found VolumeReplicationGroup with matching labels       {"pvc": "envtest-ns-sdixm/pvc-of-pv0001", "vrg": "vrg-sdixm", "labeled": ""}
  2023-11-08T21:23:29.218+0530  INFO    controllers.VolumeReplicationGroup.vrginstance  Done uploading PV/PVC cluster data to 1 of 1 S3 profile(s): [s3profile0]        {"VolumeReplicationGroup": "envtest-ns-sdixm/vrg-sdixm", "rid": "d5b10778-cf83-48ca-897c-d810ed18bea0", "State": "primary"}
  2023-11-08T21:23:29.218+0530  INFO    controllers.VolumeReplicationGroup.vrginstance  Successfully processed VolumeReplication for PersistentVolumeClaim      {"VolumeReplicationGroup": "envtest-ns-sdixm/vrg-sdixm", "rid": "d5b10778-cf83-48ca-897c-d810ed18bea0", "State": "primary", "pvc": "envtest-ns-sdixm/pvc-of-pv0001"}

**Comment by raghavendra-talur** [comment]
Closing this pull request. Will revisit later if required.

**Pull request closed without merging**